### ISSUE TITLE
rookd --ceph-config-override should accept all settings

### ISSUE THREAD

**Issue by bzub** [rook/rook#585]
I tried to set a "bdev" setting with `--ceph-config-override` but it never shows up in the merged `rookcluster.config`.  Other settings in my config file do get merged in.  Maybe rookd only looks for certain setting subsystems?

For example, these do not work:
```ini
[global]
bdev_aio_max_queue_depth = 1024
```
```ini
[bdev]
bdev aio max queue depth = 1024
```

**Comment by jbw976** [comment]
That's unexpected @bzub, the code just does a simple append of the entire config file override content, it doesn't look for specific sections: https://github.com/rook/rook/blob/master/pkg/cephmgr/mon/config.go#L158

Here's the source for that Append function: https://github.com/go-ini/ini/blob/master/ini.go#L382

Do you have a full example of the config override file you are seeing this with? I'd like to try to reproduce locally.

**Comment by bzub** [comment]
I now realize rookd wasn't creating a configuration file.  It was just using the existing `rookcluster.config`.  I tried moving that config file away but that only caused the OSD daemon to fail to start due to missing configuration options.

For some reason I had the impression that `rookcluster.config` was regenerated every time rookd daemons start.  Maybe that's not the case?  I got it to regenerate the config file after I deleted the OSD completely and `rm -rf /var/lib/rook` on the storage node.  Just deleting the OSD and its data directory didn't seem to cut it.
